Library/Homebrew/brew.rb

```diff
@@ -141,12 +145,18 @@ class MissingEnvironmentVariables < RuntimeError; end
 rescue BuildError => e
   Utils::Analytics.report_build_error(e)
   e.dump
+
+  head_unsupported_error if Homebrew.args.HEAD?
+
   exit 1
 rescue RuntimeError, SystemCallError => e
   raise if e.message.empty?
 
   onoe e
   $stderr.puts e.backtrace if ARGV.debug?
+
+  head_unsupported_error if Homebrew.args.HEAD?
```

Comment:
I'm not sure this case is needed? Is `BuildError` above insufficient?

Reply:
Without this, `brew install --HEAD mtr` which fails due to a patch didn't trigger this message shown. Patches are not `BuildError`, apparently.

Reply:
> Patches are not `BuildError`, apparently.

That might be worth changing but 🆒 to just leave this code as-is for now.